subsys/mgmt/mcumgr/transport/src/smp_shell.c

```diff
@@ -64,8 +64,9 @@ extern struct shell_transport shell_transport_uart;
 static void smp_shell_input_timeout_handler(struct k_timer *timer)
 {
 	ARG_UNUSED(timer);
-	struct shell_uart *sh_uart = (struct shell_uart *)shell_transport_uart.ctx;
-	struct smp_shell_data *const data = &sh_uart->ctrl_blk->smp;
+	struct shell_uart_int_driven *sh_uart =
+		(struct shell_uart_int_driven *)shell_transport_uart.ctx;
+	struct smp_shell_data *const data = &sh_uart->common.smp;
```

Comment:
What about adding a `shell_uart_smp_shell_data_get(void)` instead of exposing all those structures?

Reply:
#65217